src/internal/operations/backup.go

```diff
@@ -632,23 +633,98 @@ func getNewPathRefs(
 	return newPath, newLoc, updated, nil
 }
 
-func mergeDetails(
+// mergeDetailsFromAssistBackups merges the deets from the assist backups into
+// the current deets.
+func mergeDetailsFromAssistBackups(
 	ctx context.Context,
 	detailsStore streamstore.Streamer,
-	backups []kopia.BackupEntry,
-	dataFromBackup kopia.DetailsMergeInfoer,
 	deets *details.Builder,
-	writeStats *kopia.BackupStats,
+	assistBackups []kopia.BackupEntry,
 	serviceType path.ServiceType,
 	errs *fault.Bus,
-) error {
-	detailsModel := deets.Details().DetailsModel
+) (map[string]details.Entry, error) {
+	// If the backup operation did not produce any new details, or if we don't
+	// have any assist backups, we don't need to merge anything.
+	if deets.Empty() || len(assistBackups) == 0 {
+		logger.Ctx(ctx).Infow("no assist backups to merge",
+			"deets_empty", deets.Empty(),
+			"num_assist_backups", len(assistBackups))
+
+		return nil, nil
+	}
 
-	// getting the values in writeStats before anything else so that we don't get a return from
-	// conditions like no backup data.
-	writeStats.TotalNonMetaFileCount = len(detailsModel.FilterMetaFiles().Items())
-	writeStats.TotalNonMetaUploadedBytes = detailsModel.SumNonMetaFileSizes()
+	repoRefToAssistDeets := make(map[string]details.Entry)
+
+	for _, assistBackup := range assistBackups {
+		mctx := clues.Add(ctx, "assist_backup_id", assistBackup.ID)
 
+		baseDeets, err := getDetailsFromBackup(
+			mctx,
+			assistBackup.Backup,
+			detailsStore,
+			errs)
+		if err != nil {
+			return nil, clues.New("fetching details for assist backup")
+		}
+
+		for _, entry := range baseDeets.Items() {
+			rr, err := path.FromDataLayerPath(entry.RepoRef, true)
+			if err != nil {
+				return nil, clues.New("parsing base item info path").
+					WithClues(mctx).
+					With("repo_ref", path.NewElements(entry.RepoRef))
+			}
+
+			// Although this assist base has an entry it may not be the most recent.
+			// Check the reasons a snapshot was returned to ensure we only choose the
+			// recent entries.
+			if !matchesReason(assistBackup.Reasons, rr) {
+				continue
+			}
+
+			repoRefToAssistDeets[entry.RepoRef] = *entry
+		}
+	}
+
+	// Walk through new deets and update items from assist deets if needed.
+	// There are 2 possibilities here.
+	// 1. The item didn't exist in assist deets. Do nothing.
+	// 2. The item existed in the assist deets. Source it from assist deets
+	// if mod time hasn't changed. Otherwise current deets have the most
+	// recent info, do nothing.
```

Comment:
> Source it from assist deets
> if mod time hasn't changed. Otherwise current deets have the most
> recent info, do nothing.

I think this is where stuff starts to get a little fuzzy and I'm not sure all the logic will play out how we'd like. Again, this goes back to whether we assume backups can be concurrent or not

If we assume that backups aren't concurrent then this logic will work out just fine

However, if we assume that backups can be concurrent then it's possible the older backup (didn't have the Reason for this item) has the newer version of the item but we didn't select it because of the check on 681

Reply:
Okay, let me think a bit more about concurrent backups with overlapping reasons.